torchdata/datapipes/iter/util/cacheholder.py

```diff
@@ -185,13 +194,19 @@ def __init__(
 
         if filepath_fn is not None:
             _check_unpickable_fn(filepath_fn)
-        filepath_fn = _generator_to_list(filepath_fn) if inspect.isgeneratorfunction(filepath_fn) else filepath_fn
+        assert not inspect.isgeneratorfunction(filepath_fn)  # BC breaking, now only str is accepted as return
```

Comment:
Then, we need to change the dataset implementation from TorchText side. e.g.: https://github.com/pytorch/text/blob/ff1fdfce8ac030a11638b2d94d54364144586253/torchtext/datasets/imdb.py#L107